### ISSUE TITLE
TraceId and SpanId are missing from the logs

### ISSUE THREAD

**Issue by KrishnaKole** [serilog/serilog-sinks-splunk#195]
I am using v4.0.3 of this sink with below setup:
```csharp
        Log.Logger = new LoggerConfiguration()
            .Destructure.UsingAttributes()
            .Enrich.FromLogContext()
            .Enrich.WithExceptionDetails()
            .Enrich.WithCorrelationId(StaticConfig.CorrelationIdHeader, addValueIfHeaderAbsence: true)
            .WriteTo.Conditional(
                _ => !builder.Environment.IsDevelopment(),
                wt => wt.EventCollector(
                    splunkConfig.BaseUri,
                    splunkConfig.Token,
                    "services/collector/event",
                    splunkConfig.AppSource ?? Assembly.GetEntryAssembly()?.GetName().Name ?? string.Empty,
                    sourceType: "",
                    splunkConfig.AppHost ?? Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME", EnvironmentVariableTarget.Process) ?? Environment.MachineName,
                    splunkConfig.Index
                )
            )
            .WriteTo.Conditional(
                _ => builder.Environment.IsDevelopment(),
                wt => wt.Console(
                    outputTemplate:
                    "[{Timestamp:yyyy.MM.dd HH:mm:ss} {TraceId} {SpanId} {CorrelationId} {Level:u3}] {Message:lj} {Properties:j} {NewLine}{Exception}"
                )
            )
            .ReadFrom.Configuration(builder.Configuration)
            .CreateLogger();
       builder.Host.UseSerilog();
```

I get the TraceId and SpanId in console logs, but they are missing from the logs
![image](https://github.com/serilog-contrib/serilog-sinks-splunk/assets/6912890/ae19ee90-f55e-438e-ba1e-b6b0c4489c82)

Am I doing something wrong?

**Comment by EEParker** [comment]
I added a reproduction case #196. I looks like this is related to some internal changes to dotnet 6.0 and up, see https://github.com/dotnet/runtime/issues/34305

I found a related MR to serilog here: https://github.com/serilog/serilog/pull/1955/files#diff-b111b459384a29d3e58a068b0b47c05777172c9974217fb615a6976c70841de7

**Comment by EEParker** [comment]
![image](https://github.com/serilog-contrib/serilog-sinks-splunk/assets/15874076/1011e457-f037-4db2-baa5-bf3b1f372b09)
This will be fixed with 4.0.4

**Comment by EEParker** [comment]
@KrishnaKole once [4.0.4](https://www.nuget.org/packages/Serilog.Sinks.Splunk/4.0.4) is available on nuget, can you confirm this is fixed?

**Comment by KrishnaKole** [comment]
Yes, it is fixed with 4.0.4. Thank you!
